edx_lint/pylint/annotations_check.py

```diff
@@ -427,6 +446,25 @@ def check_annotation_group(self, search, annotations, node):
                 line=line_number,
             )
 
+    @utils.check_messages(MISSING_ANNOTATION)
+    def visit_call(self, node):
+        """
+        Check for missing annotations.
+        """
+        toggle_func_names = ["WaffleFlag", "NonNamespacedWaffleFlag", "WaffleSwitch", "NonNamespacedWaffleSwitch"]
+        if not isinstance(node.func, Name) or node.func.name not in toggle_func_names or not node.args:
+            return
+        toggle_name_arg = node.args[0]
+        if not isinstance(toggle_name_arg, Const) or not isinstance(toggle_name_arg.value, str):
+            return
+        toggle_name = toggle_name_arg.value
+        if toggle_name not in self.current_module_annotated_toggle_names:
```

Comment:
I've tried to think about this long and hard, and did not come up with good ideas... We can try to brainstorm, but I'm feeling pessimistic. I don't want to sound overly negative; it's just that I don't like solutions which would force developers to use toggles in one single way.

Reply:
Let's discuss when we can meet.  I'm not clear on the connection between linting to ensure a toggle is annotated, and your comment: "force developers to use toggles in one single way".

I don't feel my suggestions change the use of the toggles at all.  It would just capture more toggles that are both not annotated and don't have a simple string for a name. We can discuss my two proposals and what you don't like about them or why you don't think they will work.

Reply:
@robrap as discussed, please check the new implementation, where we check whether annotations are present before the toggle object instantation. For reference, this new linter reports the following errors on edx-platform:

```
$ pylint --load-plugins=edx_lint.pylint --disable=all --enable=toggle-missing-annotation ../edx-platform/lms ../edx-platform/cms ../edx-platform/openedx ../edx-platform/common/
************* Module lms.djangoapps.course_home_api.toggles
../edx-platform/lms/djangoapps/course_home_api/toggles.py:12:28: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/course_home_api/toggles.py:14:38: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/course_home_api/toggles.py:16:40: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module lms.djangoapps.experiments.flags
../edx-platform/lms/djangoapps/experiments/flags.py:79:12: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module lms.djangoapps.experiments.tests.test_flags
../edx-platform/lms/djangoapps/experiments/tests/test_flags.py:41:20: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/experiments/tests/test_flags.py:108:22: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/experiments/tests/test_flags.py:166:15: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/experiments/tests/test_flags.py:176:24: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/experiments/tests/test_flags.py:179:27: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/experiments/tests/test_flags.py:181:26: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/experiments/tests/test_flags.py:184:29: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module lms.djangoapps.instructor_task.config.waffle
../edx-platform/lms/djangoapps/instructor_task/config/waffle.py:27:53: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/lms/djangoapps/instructor_task/config/waffle.py:32:52: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module cms.djangoapps.export_course_metadata.toggles
../edx-platform/cms/djangoapps/export_course_metadata/toggles.py:15:30: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module cms.djangoapps.models.settings.waffle
../edx-platform/cms/djangoapps/models/settings/waffle.py:11:31: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module cms.djangoapps.contentstore.views.videos
../edx-platform/cms/djangoapps/contentstore/views/videos.py:83:33: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module cms.djangoapps.contentstore.config.waffle
../edx-platform/cms/djangoapps/contentstore/config/waffle.py:33:28: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/cms/djangoapps/contentstore/config/waffle.py:39:25: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module openedx.features.course_experience
../edx-platform/openedx/features/course_experience/__init__.py:20:35: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/features/course_experience/__init__.py:25:34: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/features/course_experience/__init__.py:30:27: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/features/course_experience/__init__.py:48:34: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/features/course_experience/__init__.py:51:27: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/features/course_experience/__init__.py:69:39: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module openedx.features.effort_estimation.toggles
../edx-platform/openedx/features/effort_estimation/toggles.py:16:34: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module openedx.core.djangoapps.waffle_utils.tests.test_init
../edx-platform/openedx/core/djangoapps/waffle_utils/tests/test_init.py:37:23: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/core/djangoapps/waffle_utils/tests/test_init.py:85:27: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/core/djangoapps/waffle_utils/tests/test_init.py:111:27: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/core/djangoapps/waffle_utils/tests/test_init.py:123:27: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module openedx.core.djangoapps.waffle_utils.tests.test_views
../edx-platform/openedx/core/djangoapps/waffle_utils/tests/test_views.py:19:19: E7664: missing feature toggle annotation (toggle-missing-annotation)
../edx-platform/openedx/core/djangoapps/waffle_utils/tests/test_views.py:119:22: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module openedx.core.djangoapps.models.config.waffle
../edx-platform/openedx/core/djangoapps/models/config/waffle.py:23:47: E7664: missing feature toggle annotation (toggle-missing-annotation)
************* Module openedx.core.djangoapps.schedules.config
../edx-platform/openedx/core/djangoapps/schedules/config.py:34:32: E7664: missing feature toggle annotation (toggle-missing-annotation)
```

Reply:
Looks great.